example-2/evm/src/contracts/lendingHub/HubSetters.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "./HubState.sol";
+import "../HubSpokeStructs.sol";
+import "./HubGetters.sol";
+
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+import "@pythnetwork/pyth-sdk-solidity/MockPyth.sol";
+
+contract HubSetters is HubSpokeStructs, HubState, HubGetters {
+    function setChainId(uint16 chainId) internal {
+        _state.provider.chainId = chainId;
+    }
+
+    function setWormhole(address wormholeAddress) internal {
+        _state.provider.wormhole = payable(wormholeAddress);
+    }
+
+    function setTokenBridge(address tokenBridgeAddress) internal {
+        _state.provider.tokenBridge = tokenBridgeAddress;
+    }
+
+    function setPyth(address pythAddress) internal {
+        _state.provider.pyth = IPyth(pythAddress);
+    }
+
+    function setOracleMode(uint8 oracleMode) internal {
+        _state.oracleMode = oracleMode;
+    }
+
+    function setConsistencyLevel(uint8 consistencyLevel) internal {
+        _state.consistencyLevel = consistencyLevel;
+    }
+
+    function registerSpokeContract(uint16 chainId, address spokeContractAddress) internal {
+        _state.spokeContracts[chainId] = spokeContractAddress;
+    }
+
+    function registerAssetInfo(address assetAddress, AssetInfo memory info) internal {
+        uint256[] memory kinks = info.interestRateModel.kinks;
+        uint256[] memory rates = info.interestRateModel.rates;
+
+        uint n = kinks.length;
+        uint m = rates.length;
+
+        require(n == m, "lengths of kinks and rates arrays don't match");
+        require(kinks[0]==0, "first kink must be at 0");
+
+        for(uint i=1; i < n; i++) {
+            require(kinks[i] > kinks[i-1], "kinks must be monotonically increasing");
+        }
+
+        require(kinks[n-1]==info.interestRateModel.ratePrecision, "last kink must be 1 (i.e. ratePrecision)");
+
+        for(uint i=1; i < m; i++) {
+            require(rates[i] >= rates[i-1], "rates must be monotonically non-decreasing");
+        }
+
+        _state.assetInfos[assetAddress] = info;
+
+        AccrualIndices memory accrualIndices;
+        accrualIndices.deposited = 1 * getInterestAccrualIndexPrecision();
+        accrualIndices.borrowed = 1 * getInterestAccrualIndexPrecision();
```

Comment:
Were these `1` values going to be replaced by something? Should probably take this out